app/controllers/api/v1/workflows_controller.rb

```diff
@@ -82,15 +83,13 @@ def post_link_actions(workflow)
     if workflow.set_member_subjects.exists?
       case relation
       when :retired_subjects, 'retired_subjects'
-        WorkflowRetiredCountWorker.perform_async(workflow.id)
-
         params[:retired_subjects].each do |subject_id|
           NotifySubjectSelectorOfRetirementWorker.perform_async(subject_id, workflow.id)
         end
       when :subject_sets, 'subject_sets'
+        CalculateProjectCompletenessWorker.perform_async(workflow.project_id)
```

Comment:
There is no guarantee that this will run before the retired count worker unfortunately.